### ISSUE TITLE
kubeadm cannot join control plane node via --config flag

### ISSUE THREAD

**Issue by stv0g** [kubernetes/kubernetes#89499]
**What happened**: kubeadm currently fails to join a control plane node in case all settings are provided via a kubeadm configuration file.

**What you expected to happen**: Joining a new control plane node should be possible without the `--control-plane` flag

**How to reproduce it (as minimally and precisely as possible)**:

Given is the following configuration file:
```yaml
---
apiVersion: kubeadm.k8s.io/v1beta2
kind: JoinConfiguration
discovery:
  bootstrapToken:
    apiServerEndpoint: "api.k8s:6443"
    token: xxxxxxxxxxxxxx
    cACertHashes: [ ]
    unsafeSkipCAVerification: true

controlPlane:
  certificateKey: yyyyyyyyyyyyyyyyyy
```

When joining a node via: `kubeadm join --config kubeadm-join-config.yaml

I get the following error:
```
[preflight] WARNING: JoinControlPane.controlPlane settings will be ignored when control-plane flag is not set.
```

Adding the `--control-plane` flag: `kubeadm join --control-plane --config kubeadm-join-config.yaml` yields:

```
can not mix '--config' with arguments [control-plane]
To see the stack trace of this error execute with --v=5 or higher
```

**Anything else we need to know?**:

Relevant code locations:
- https://github.com/kubernetes/kubernetes/blob/06ad960bfd03b39c8310aaf92d1e7c12ce618213/cmd/kubeadm/app/cmd/join.go#L344
- https://github.com/kubernetes/kubernetes/blob/06ad960bfd03b39c8310aaf92d1e7c12ce618213/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L452

**Environment**:
- Kubernetes version: v1.17.3 (06ad960bfd03b39c8310aaf92d1e7c12ce618213)
- Cloud provider or hardware configuration: baremetal, x86_86, 768GB DDR, 2x 16C/32T Xeon Gold
- OS: CentOS 8.1 
- Kernel: Linux node1.k8s.jupyter.rwth-aachen.de 5.5.11-1.el8.elrepo.x86_64 #1 SMP Fri Mar 20 10:38:49 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm
- Network plugin and version (if this is a network-related bug): none

**Comment by stv0g** [comment]
This seems like an easy fix. If nobody objects, I would remove the check here:

https://github.com/kubernetes/kubernetes/blob/06ad960bfd03b39c8310aaf92d1e7c12ce618213/cmd/kubeadm/app/cmd/join.go#L344

**Comment by neolit123** [comment]
the warning is false; your config will still be respected.
we are tracking the issue here:
https://github.com/kubernetes/kubeadm/issues/2065

and a fix will be available in 1.18.1, but cannot be backported to 1.17*

/close
